src/main/java/in/projecteka/gateway/common/Validator.java

```diff
@@ -27,24 +26,30 @@
     CacheAdapter<String, String> requestIdMappings;
 
 
-    public Mono<ValidatedRequest> validateRequest(HttpEntity<String> requestEntity) {
-        List<String> xHipIds = requestEntity.getHeaders().get(X_HIP_ID);
-        if (xHipIds == null || xHipIds.isEmpty()) {
-            return Mono.error(ClientError.hipIdMissing());
+    public Mono<ValidatedRequest> validateRequest(HttpEntity<String> requestEntity, String id) {
+        List<String> ids = requestEntity.getHeaders().get(id);
+        if (ids == null || ids.isEmpty()) {
+            return Mono.error(ClientError.idMissingInHeader(id));
         }
-        String xHipId = xHipIds.get(0);
-        Optional<YamlRegistryMapping> hipConfig = bridgeRegistry.getConfigFor(xHipId, ServiceType.HIP);
-        if (hipConfig.isEmpty()) {
-            return Mono.error(ClientError.mappingNotFoundForHipId());
+        String xid = ids.get(0);
+        Optional<YamlRegistryMapping> config = bridgeRegistry.getConfigFor(xid, ServiceType.HIP);
+        if(id.equals(X_CM_ID)){
+            config = cmRegistry.getConfigFor(xid);
```

Comment:
you can have a check for id equals CM_ID/HIP before getting the config for HIP .. you can have a ternary check .. for ex:  
Optional<YamlRegistryMapping> config;
config = id.equals(X_HIP_ID) ? bridgeRegistry.getConfigFor(xid, ServiceType.HIP) : cmRegistry.getConfigFor(xid);

Reply:
But happens if its a X_HIU_ID? Then we can't really use ternary check.

Reply:
If that is the case, I think you should make the param ServiceType.HIP generic one right?

Reply:
I don't get when you say generic. There will be an other ServiceType.HIU in that case and that should do.

Reply:
I'm just saying in future if we might get HIU_ID as a header, then also we must have a check before computing getConfigFor. So.. for now we can use this:
config = id.equals(X_CM_ID) ? cmRegistry.getConfigFor(xid) : bridgeRegistry.getConfigFor(xid, ServiceType.HIP) ;

Correct me if I'm wrong.